### DISCUSSION TITLE
Howto install BondLib

### DISCUSSION THREAD

**Discussion by Robbty** [OpenModelica/OpenModelica#11152] [Q&A]
Hi,
I’m new with OpenModelica and I have installed OMEdit with MSL 4.0. Now I want to install the BondLib. I find out that the current OMEdit do not support MSL 2.2 which is necessary for using the BondLib.
Is there anyone who can tell me step by step how to install to run the BondLib?
I read many day‘s a lot of documentations and books to learn handling OpenModelica, and also I searched a lot in the net but I did not find any install doc and so it was not yet possible to install a running version of the BondLib.

It would be very great if anyone can support me to successfully install it.

Thank you in advance,
Peter

**Comment by casella** [comment]
I'm sorry, but OpenModelica doesn't support any library that uses MSL < 3.2.1.

The reason is that previous versions of the MSL had fundamental issues, either because they were not complying to the Modelica Language Specification, or because the Modelica Language Specification was actually flawed. The Modelica Association fixed the language with versions 3.2 rev1 and rev2, and later with 3.3 rev1, and fixed MSL 3.2 with 3.2.1, which had the same models but was now truly compliant with the language definition.

OpenModelica has always strived to implement the standard without compromise. It supported part of MSL 2.x in the very old days, but it cannot really fully support it (e.g. the MultiBody models), because of these issues.

If you want to use a MSL 2.2 compliant library with OpenModelica, you have to first convert it to MSL 3.x or, even better, to MSL 4.0.0. You can do it e.g. using Dymola (this was possible in the past even with the demo version, not sure if this is still the case).

**Reply by Robbty** [reply]
Hi casella,
Thank you very much for your response. I will try out converting it using the demo version of Dymola and give a feedback of the result.
